src/hotspot/share/gc/shared/memAllocator.hpp

```diff
@@ -118,4 +118,35 @@ class ClassAllocator: public MemAllocator {
   virtual oop initialize(HeapWord* mem) const;
 };
 
+// Manages a scope where a failed heap allocation results in
+// a shared, backtrace-less OOME instance being thrown.
+// Also, JVMTI resource exhausted events and handling of
+// -XX:+HeapDumpOnOutOfMemoryError and -XX:OnOutOfMemoryError
```

Comment:
This is my main concern with this PR: I don't think disabling JVMTI events and `-XX:+HeapDumpOnOutOfMemoryError` and `-XX:OnOutOfMemoryError` handling is correct except for the specific use in the `JVMCIRuntime::new_*` routines.